management-ui/server.py

```diff
@@ -144,34 +281,44 @@ async def cmd_delete_session(provisioning_session_id: str, config: Configuration
     
     return JSONResponse(content={"message": f"Provisioning Session {provisioning_session_id} and all its resources were destroyed"}, status_code=200)
 
-"""
-Endpoint: Create Content Hosting Configuration from JSON data example
-HTTP Method: POST
-Path: /set_stream/{provisioning_session_id}
-Description: This endpoint will add Content hosting configuration to particular provisioning session, taking JSON example data.
-"""
 @app.post("/set_stream/{provisioning_session_id}")
 async def set_stream(provisioning_session_id: str, config: Configuration = Depends(get_config)):
-    session = await get_session(config)
-    json_path = "examples/ContentHostingConfiguration_Llama-Drama_pull-ingest.json"
-    with open(json_path, 'r') as f:
-        chc = json.load(f)
-    
-    old_chc = await session.contentHostingConfigurationGet(provisioning_session_id)
-    
-    if old_chc is None:
-        result = await session.contentHostingConfigurationCreate(provisioning_session_id, chc)
-    else:
-        for dc in chc['distributionConfigurations']:
-            for strip_field in ['canonicalDomainName', 'baseURL']:
-                if strip_field in dc:
-                    del dc[strip_field]
-        result = await session.contentHostingConfigurationUpdate(provisioning_session_id, chc)
+    try:
+        # Retrieve the session using MediaConfiguration
+        media_configuration = await initialize_media_configuration()
+        media_session = await media_configuration.mediaSessionByProvisioningSessionId(provisioning_session_id)
 
-    if not result:
+        if media_session is None:
+            raise HTTPException(status_code=404, detail="Provisioning session not found")
+
+        # Perform the content hosting configuration
+        json_path = "examples/ContentHostingConfiguration_Llama-Drama_pull-ingest.json"
+        with open(json_path, 'r') as f:
+            chc = json.load(f)
+
+        #session = await get_session(config)
+        session = await get_session()
+        old_chc = await session.contentHostingConfigurationGet(provisioning_session_id)
+
+        if old_chc is None:
+            result = await session.contentHostingConfigurationCreate(provisioning_session_id, chc)
+        else:
+            for dc in chc['distributionConfigurations']:
+                for strip_field in ['canonicalDomainName', 'baseURL']:
+                    if strip_field in dc:
+                        del dc[strip_field]
+            result = await session.contentHostingConfigurationUpdate(provisioning_session_id, chc)
```

Comment:
For the new operation this should alter the `media_session.media_entry` and then `media_session.configuration.synchronise()` to push the changes to the AF. You shouldn't be doing `M1Session` operations directly anymore.